src/stream/min_by.rs

```diff
@@ -0,0 +1,54 @@
+use std::cmp::Ordering;
+use std::pin::Pin;
+
+use super::stream::Stream;
+use crate::future::Future;
+use crate::task::{Context, Poll};
+
+/// A future that yields the minimum item in a stream by a given comparison function.
+#[derive(Clone, Debug)]
+pub struct MinBy<S: Stream, F> {
```

Comment:
Since `MinBy` is here in a separate file, it'd be a good idea to move `Take` and others into their own files, too.

Reply:
But we can do that in a follow-up PR :)

Reply:
See https://github.com/async-rs/async-std/pull/150!